### ISSUE TITLE
label propagation to nodes doesnt happen when using aws as the cloud provider in k8s

### ISSUE THREAD

**Issue by aemneina** [rancher/rancher#10506]
**Rancher versions:**
rancher/server: 1.6.10
rancher/agent: 1.2.6

**Steps to Reproduce:**
1. deploy a k8s environment that uses aws as the cloud provider
2. use the ec2 driver to launch hosts, add labels to hosts (understandably this isnt supported officially, it seems to mostly work. And the ability to use machine drivers would rock)

**Results:**
no labels get propagated to the k8s nodes
Following errors seen in rancher/kubernetes-agent logs:

```
12/7/2017 3:52:30 PMtime="2017-12-07T23:52:30Z" level=info msg="Skipping event: [ADDED] for namespace: kube-system"
12/7/2017 3:52:30 PMtime="2017-12-07T23:52:30Z" level=info msg="Received event: [ADDED] for Service: default/kubernetes, Handling event."
12/7/2017 3:52:30 PMtime="2017-12-07T23:52:30Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:30 PMtime="2017-12-07T23:52:30Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:30 PMtime="2017-12-07T23:52:30Z" level=info msg="Received event: [ADDED] for Service: kube-system/tiller-deploy, Handling event."
12/7/2017 3:52:31 PMtime="2017-12-07T23:52:31Z" level=info msg="Received event: [ADDED] for Service: kube-system/kubernetes-dashboard, Handling event."
12/7/2017 3:52:31 PMtime="2017-12-07T23:52:31Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:31 PMtime="2017-12-07T23:52:31Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:32 PMtime="2017-12-07T23:52:32Z" level=info msg="Received event: [ADDED] for Service: kube-system/monitoring-grafana, Handling event."
12/7/2017 3:52:32 PMtime="2017-12-07T23:52:32Z" level=info msg="Received event: [ADDED] for Service: kube-system/monitoring-influxdb, Handling event."
12/7/2017 3:52:32 PMtime="2017-12-07T23:52:32Z" level=info msg="Received event: [ADDED] for Service: kube-system/heapster, Handling event."
12/7/2017 3:52:32 PMtime="2017-12-07T23:52:32Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:32 PMtime="2017-12-07T23:52:32Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:33 PMtime="2017-12-07T23:52:33Z" level=info msg="Received event: [ADDED] for Service: kube-system/kube-dns, Handling event."
12/7/2017 3:52:33 PMtime="2017-12-07T23:52:33Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:33 PMtime="2017-12-07T23:52:33Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:35 PMtime="2017-12-07T23:52:35Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:35 PMtime="2017-12-07T23:52:35Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:36 PMtime="2017-12-07T23:52:36Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:36 PMtime="2017-12-07T23:52:36Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:38 PMtime="2017-12-07T23:52:38Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"
12/7/2017 3:52:38 PMtime="2017-12-07T23:52:38Z" level=error msg="Error getting node: [sank8saws] by name from kubernetes, err: [nodes \"sank8saws\" not found]"```

**Comment by ppadala** [comment]
@aemneina, any ETA on this?

**Comment by moelsayed** [comment]
Tested with the following parameters:
- rancher server v1.6.15-rc5
- created a k8s environment with aws cloud provider
- created 15 hosts, in 3, 4, 8 batches, each with a different set of labels. 
- in all cases the label propagated correctly. 

I was not able to reproduce this issue with latest RC.

**Comment by moelsayed** [comment]
The problem is with node names. 

K8s defaults to use aws internal instance hostnames, while rancher uses "host names" defined in while creating hosts. 

I tried to update `--override-hostname` with kubelet to use the rancher host names with k8s, but using `--cloud-provider=aws` overrides `--override-hostname`.

I will try to fix this in `kubernetes-agent`.

**Comment by deniseschannon** [comment]
Fix is with `kubernetes-agent:v0.6.7`

**Comment by soumyalj** [comment]
Tested with master 1.6-development.
Created a K8s environment with AWS hosts and added labels while adding hosts.
`kubectl describe node nodename` showed the label correctly
Label was added after the nodes were created and labels were displayed correctly
```
LAPTOP-6CQEFUBT:~ soumya$ kubectl describe node soumyalabelpropagation1
Name:			soumyalabelpropagation1
Role:			
Labels:			awstest=value
			beta.kubernetes.io/arch=amd64
			beta.kubernetes.io/instance-type=rancher
			beta.kubernetes.io/os=linux
			failure-domain.beta.kubernetes.io/region=Region1
			failure-domain.beta.kubernetes.io/zone=FailureDomain1
			io.rancher.host.os=linux
			io.rancher.host.provider=aws
			io.rancher.host.region=us-west-1
			io.rancher.host.zone=us-west-1b
			kubernetes.io/hostname=soumyalabelpropagation1
			labelagain=test



```
